evalml/tests/automl_tests/test_automl_iterative_algorithm.py

```diff
@@ -0,0 +1,1146 @@
+    with env.test_context(score_return_value={"Log Loss Binary": 1.0, "F1": 0.5}):
+        automl.search()
+
+    assert start_iteration_callback.call_count == 2
+    proposed_parameters = start_iteration_callback.call_args_list[1][0][0].parameters
+    assert proposed_parameters.keys() == {
+        "RF Classifier Select From Model",
+        "Logistic Regression Classifier",
+    }
+    assert (
+        proposed_parameters["RF Classifier Select From Model"]["number_features"]
+        == X.shape[1]
+    )
+
+
+def test_automl_allowed_component_graphs_iterative_algorithm(
+    dummy_classifier_estimator_class,
+    X_y_binary,
+):
+    X, y = X_y_binary
+    allowed_component_graphs = {
+        "Mock Binary Classification Pipeline": [dummy_classifier_estimator_class]
+    }
+    aml = AutoMLSearch(
+        X_train=X,
+        y_train=y,
+        problem_type="binary",
+        allowed_component_graphs=allowed_component_graphs,
+        max_iterations=10,
+        _automl_algorithm="iterative",
+    )
+
+    assert aml._automl_algorithm.allowed_component_graphs == allowed_component_graphs
+
+
+def test_component_graph_with_incorrect_problem_type_iterative(
+    dummy_classifier_estimator_class, X_y_binary
+):
+    X, y = X_y_binary
+    with pytest.raises(ValueError, match="not valid for this component graph"):
+        AutoMLSearch(
+            X_train=X,
+            y_train=y,
+            problem_type="regression",
+            allowed_component_graphs={
+                "Mock Binary Classification Pipeline": [
+                    dummy_classifier_estimator_class
+                ]
+            },
+            _automl_algorithm="iterative",
+        )
+
+
+@pytest.mark.parametrize("return_dict", [True, False])
+def test_describe_pipeline_with_ensembling_iterative(
+    return_dict, X_y_binary, AutoMLTestEnv, caplog
+):
+    X, y = X_y_binary
+    two_stacking_batches = 1 + 2 * (len(get_estimators(ProblemTypes.BINARY)) + 1)
+    automl = AutoMLSearch(
+        X_train=X,
+        y_train=y,
+        problem_type="binary",
+        max_batches=two_stacking_batches,
+        objective="Log Loss Binary",
+        ensembling=True,
+        optimize_thresholds=False,
+        error_callback=raise_error_callback,
+        _automl_algorithm="iterative",
+    )
+
+    score_side_effect = [
+        {"Log Loss Binary": score}
+        for score in np.arange(
+            0, -1 * automl.max_iterations * automl.data_splitter.get_n_splits(), -0.1
+        )
+    ]  # Dcreases with each call
+
+    test_env = AutoMLTestEnv("binary")
+    with test_env.test_context(mock_score_side_effect=score_side_effect):
+        automl.search()
+    pipeline_names = automl.rankings["pipeline_name"]
+    assert pipeline_names.str.contains("Ensemble").any()
+
+    ensemble_ids = [
+        _get_first_stacked_classifier_no() - 1,
+        len(automl.results["pipeline_results"]) - 1,
+    ]
+
+    for i, ensemble_id in enumerate(ensemble_ids):
```

Comment:
This uses tests the number of batches/pipelines used in `IterativeAlgorithm` to hit the number required for ensembling to turn on.